src/main/java/pingis/services/DataLoader.java

```diff
@@ -11,117 +13,111 @@
 
 @Component
 public class DataLoader implements ApplicationRunner {
-//CHECKSTYLE:OFF
     private ChallengeRepository cr;
     private TaskRepository tr;
+    private LinkedHashMap<Challenge, ArrayList<Task>> challenges;
```

Comment:
This makes it possible for us to loop through the HashMap structure in predictable ordering. 
* In regular ´´´HashMap´´´ the keys are not ordered in terms of their addition order. 
* By using ´´´LinkedHashMap´´´ we can loop and access the keys in the same order we put them into the data structure. This is helpful for the Challenges' id's.